src/state/actions/dispatchArNSUpdate.ts

```diff
@@ -3,140 +3,214 @@ import {
 import { captureException } from '@sentry/react';
 import { AoAddress } from '@src/types';
 import { NETWORK_DEFAULTS } from '@src/utils/constants';
 import eventEmitter from '@src/utils/events';
-import { queryClient } from '@src/utils/network';
+import { buildAntStateQuery, queryClient } from '@src/utils/network';
 import { Tag } from 'arweave/node/lib/transaction';
+import { pLimit } from 'plimit-lit';
 import { Dispatch } from 'react';
 
 import { ArNSAction } from '../reducers/ArNSReducer';
 
-export function dispatchArNSUpdate({
-  emitter,
+export async function dispatchArNSUpdate({
   dispatch,
   walletAddress,
   arioProcessId,
   ao,
   antAo,
   aoNetworkSettings,
 }: {
-  emitter: ArNSEventEmitter;
   dispatch: Dispatch<ArNSAction>;
   walletAddress: AoAddress;
   arioProcessId: string;
   ao: AoClient;
   antAo: AoClient;
   aoNetworkSettings: typeof NETWORK_DEFAULTS.AO;
 }) {
-  dispatch({ type: 'setDomains', payload: {} });
-  dispatch({ type: 'setAnts', payload: {} });
-  dispatch({ type: 'setPercentLoaded', payload: 0 });
-  dispatch({ type: 'setAntCount', payload: 0 });
-  dispatch({
-    type: 'setLoading',
-    payload: true,
-  });
-  emitter.on('process', async (id, process) => {
-    const handlers = await queryClient.fetchQuery({
-      queryKey: ['handlers', id, aoNetworkSettings.ANT],
-      queryFn: async () => {
-        // validate transfer supports eth addresses and single char undername fix and max ttl
-        const drySetRecordRes = await antAo
-          .dryrun({
-            process: id,
-            Owner: walletAddress.toString(),
-            From: walletAddress.toString(),
-            tags: [
-              { name: 'Action', value: 'Set-Record' },
-              { name: 'Sub-Domain', value: 'a' },
-              { name: 'Transaction-Id', value: 'test-'.padEnd(43, '1') },
-              { name: 'TTL-Seconds', value: '86400' },
-            ],
-          })
-          .catch(() => {
-            return {} as ReturnType<typeof ao.dryrun>;
-          });
+  try {
+    const limit = pLimit(20);
+    // reset queries
+    queryClient.resetQueries({
+      queryKey: ['ant'],
+    });
+    queryClient.resetQueries({
+      queryKey: ['handlers'],
+    });
+    dispatch({ type: 'reset' });
+    dispatch({ type: 'setDomains', payload: {} });
+    dispatch({ type: 'setAnts', payload: {} });
+    dispatch({ type: 'setPercentLoaded', payload: 0 });
+    dispatch({ type: 'setAntCount', payload: 0 });
+    dispatch({
+      type: 'setLoading',
+      payload: true,
+    });
+    const arioContract = ARIO.init({
+      process: new AOProcess({ processId: arioProcessId, ao }),
+    }) as AoARIORead;
+    const antRegistry = ANTRegistry.init({
+      process: new AOProcess({
+        processId: ANT_REGISTRY_ID,
+        ao: connect(aoNetworkSettings.ANT),
+      }),
+    });
+    const [arnsRecords, allUserAnts] = await Promise.all([
+      arioContract.getArNSRecords({ limit: 100_000 }),
```

Comment:
So if we have more than 100K records this won't retrieve all of them? (And if we have that many this whole scheme has bigger problems)

Reply:
FWIW: looks like with testnet it's a 192kb download each app start/refresh of table

Reply:
yes - this will be reduced significantly with the filter by process ID changes